exporter/exporterhelper/request.go

```diff
@@ -0,0 +1,39 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package exporterhelper // import "go.opentelemetry.io/collector/exporter/exporterhelper"
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/exporter/exporterhelper/internal"
+)
+
+// Request represents a single request that can be sent to the endpoint.
+type Request interface {
```

Comment:
Should we make this interface unimplementable and have some public base request struct to provide default implementations for future methods?

Reply:
Alternative is to have also `RequestItemer` (or something like that) so we just don't need an interface and we use `any`. And ask users to provide funcs that work with the request. Another option is to ask user to implement the Send on that request as well, but right now `ItemCount` is on the request but Send is not, so inconsistent.

Reply:
> Alternative is to have also RequestItemer (or something like that) so we just don't need an interface and we use `any`. And ask users to provide funcs that work with the request

In that case, do you suggest we make this an optional interface? We could do that, but the new itemized queue and batching exporter helpers won't work. We can probably allow that in case clients don't need that.

> Another option is to ask user to implement the Send on that request as well, but right now ItemCount is on the request but Send is not, so inconsistent.

What if we ask clients to implement `Request` with only one required `Send` function, and if they want itemized queue or batching helpers, they implement the optional interface `RequestItemer`. Then, we can possibly introduce bytes-sized queue or batching helpers if they implement `RequestSizer`.

```golang
type Request interface {
	Send(ctx context.Context) error
}

type RequestItemer interface {
	ItemsCount()  int
}

type RequestSizer interface {
	BytesSize() int
}
```

WDYT?

Reply:
I submitted another PR with the suggested approach